### DISCUSSION TITLE
Can py-pde solve a system of differential-algebraic equations?

### DISCUSSION THREAD

**Discussion by jAniceto** [zwicker-group/py-pde#471] [Q&A] [answered]
If I have a system of differential-algebraic equations, like:

$$\frac{\partial C}{\partial t} = f_1(C, C_p, \frac{\partial C}{\partial z})$$

$$\frac{\partial C_p}{\partial t} = f_2(C, C_p, \frac{\partial q}{\partial z})$$

$$q = f_3(C_p)$$

Dependent variables are $C$, $C_p$ and $q$. Independent are time, $t$, and position, $z$.

Am I correct that this is not possible to solve in py-pde?

**Comment by david-zwicker** [comment, marked as answer]
Generally, `py-pde` cannot solve differential-algebraic equations, but I'm not sure your equation falls under this category. Can't you simply take a derivative w.r.t. to `z` of the last equation and replace it in the second? In this case, you would be left with only two fields, both described by PDEs. If you need `q`, you could always re-create it later using the last equation.

**Reply by jAniceto** [reply]
Yes, in some cases that would be the approach but in others I have actual DAEs. Maybe my example was not the best.
Thanks again.
